pkg/cli/config.go

```diff
@@ -53,16 +55,29 @@ func NewTimelockCLI() (*Config, error) {
 
 	if os.Getenv("FROM_BLOCK") != "" {
 		fb, err := strconv.Atoi(os.Getenv("FROM_BLOCK"))
-		if err == nil {
-			c.FromBlock = int64(fb)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse FROM_BLOCK value: %w", err)
 		}
+
+		c.FromBlock = int64(fb)
 	}
 
 	if os.Getenv("POLL_PERIOD") != "" {
 		pp, err := strconv.Atoi(os.Getenv("POLL_PERIOD"))
-		if err == nil {
-			c.PollPeriod = int64(pp)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse POLL_PERIOD value: %w", err)
+		}
+
+		c.PollPeriod = int64(pp)
+	}
+
+	if os.Getenv("EVENT_LISTENER_POLL_PERIOD") != "" {
+		pp, err := strconv.Atoi(os.Getenv("EVENT_LISTENER_POLL_PERIOD"))
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse EVENT_LISTENER_POLL_PERIOD value: %w", err)
 		}
+
+		c.EventListenerPollPeriod = int64(pp)
```

Comment:
These seem like required fields now, which stops the worker from running even if they are still using websockets.

Perhaps there is an option here to add sane defaults?

Reply:
yeah, that's a good point. I think the initial version had it as optional but as I was fixing the config validations I forgot about it.

Reply:
Actually, now that I've played a bit with the implementation it seems to work as we expect.

The "startCommand" type loads the config from the environment and initializes them all with "empty values" (e.g., `eventListenerPollPeriod` is an integer so it's initialized with `0`).

Then the timelock constructor ensures that `eventListenerPollPeriod` is greater than `0` if the node url scheme is `http` or `https`.

The code you highlighted only ensures that **IF** the user set the `EVENT_LISTENER_POLL_PERIOD` environment variable, it does contain a valid number (and aborts otherwise).